clients/src/main/java/org/apache/kafka/server/authorizer/AuthorizableRequestContext.java

```diff
@@ -69,4 +69,11 @@
      * Returns the correlation id from the request header.
      */
     int correlationId();
+
+    /**
+     * Returns the initial principal name for a forwarded request.
+     */
+    default String initialPrincipalName() {
```

Comment:
Does this need to be here?  I'm concerned that having it here will eventually lead to us using it for authorization, which it shouldn't be.

Reply:
Yea, we need to have it here for audit logging. We could of course have the meta comment suggest "do not use for authorization"

Reply:
Yes, can you add a comment saying this should be used for logging, but not for authorization?